packages/legend-query/src/stores/QueryBuilderResultState.ts

```diff
@@ -80,49 +127,67 @@ export class QueryBuilderResultState {
         this.queryBuilderState.querySetupState.runtime,
         `Runtime is required to execute query`,
       );
-      let query: RawLambda;
-      if (this.queryBuilderState.isQuerySupported()) {
-        const lambdaFunction = buildLambdaFunction(this.queryBuilderState, {
-          isBuildingExecutionQuery: true,
-        });
-        query =
-          this.queryBuilderState.buildRawLambdaFromLambdaFunction(
-            lambdaFunction,
-          );
-      } else {
-        query = guaranteeNonNullable(
-          this.queryBuilderState.queryUnsupportedState.rawLambda,
-          'Lambda is required to execute query',
-        );
-        if (
-          !this.queryBuilderState.mode.isParametersDisabled &&
-          this.queryBuilderState.queryParametersState.parameters.length
-        ) {
-          const letlambdaFunction = buildParametersLetLambdaFunc(
-            this.queryBuilderState,
-          );
-          const letRawLambda =
-            this.queryBuilderState.buildRawLambdaFromLambdaFunction(
-              letlambdaFunction,
-            );
-          // reset paramaters
-          if (Array.isArray(query.body) && Array.isArray(letRawLambda.body)) {
-            letRawLambda.body = [
-              ...(letRawLambda.body as object[]),
-              ...(query.body as object[]),
-            ];
-            query = letRawLambda;
+      const query = this.buildExecutionRawLambda();
+      const result =
+        (yield this.queryBuilderState.graphManagerState.graphManager.executeMapping(
+          this.queryBuilderState.graphManagerState.graph,
+          mapping,
+          query,
+          runtime,
+          PureClientVersion.VX_X_X,
+          {
+            serializationFormat,
+          },
+        )) as ExecutionResult;
+      let contentType: ContentType;
+      let fileName = 'result';
```

Comment:
have you tested this behavior with exporting multiple times? will it cause `replace` behavior or will it create `result (1)`, `result (2)` files?

Reply:
result (2)... result(3)